### ISSUE TITLE
Split street_name into street_name and street_ref

### ISSUE THREAD

**Issue by jguthrie100** [graphhopper/graphhopper#1322]
I've been doing some work on the openstreetmap project, specifically with the routing, and its been useful to have the street_name (i.e. `"Mittelstrasse"`) separate from the street_ref (i.e. `"E13"`).

At the moment, its all in one string `"Mittelstrasse, E13"`, which is fine, but there are cases when its useful to have the two separate entities (i.e. in the case of openstreetmap, when giving the output some custom formatting. OSRM thankfully splits up the data so we can format it as we please)

I'd be happy to write a pull request that splits up the data, but it would create quite a change for the output which would probably effect quite a lot of of applications using graphhopper, so just wanted to check here first!

**Comment by devemux86** [comment]
Probably mean `EncodingManager.applyWayTags`?

**Comment by karussell** [comment]
Yes at some point we need to split this. This request is very similar to #259 from the storage perspective. Also #1086 should be thought of.

How we migrate existing users is also interesting. On the Java side we could use a deprecation notice on getName but for the json we probably have to keep it longer and use other properties to have it splitted.

**Comment by karussell** [comment]
There is some work going on to make the name index more flexible that would allow this here and a lot more (destination tag, road name translation, ...)

https://github.com/graphhopper/graphhopper/commits/string_index
